RecoLocalTracker/SiStripClusterizer/src/ThreeThresholdAlgorithm.cc

```diff
@@ -80,7 +82,8 @@ inline void ThreeThresholdAlgorithm::addToCandidate(State& state, uint16_t strip
   while (++state.lastStrip < strip)
     state.ADCs.push_back(0);  // pad holes
 
-  state.ADCs.push_back(adc);
+  if (state.ADCs.size() <= MaxClusterSize)
+    state.ADCs.push_back(adc);
```

Comment:
This is for compatibility with the GPU implementation, which has a MaxClusterSize limit so that the cluster array can be preallocated with a fixed size.